lib/std/Build/Step/InstallDir.zig

```diff
@@ -101,6 +104,21 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
 
         switch (entry.kind) {
             .directory => try cwd.makePath(dest_path),
+            .sym_link => {
+                if (!self.options.create_sym_links)
+                    continue;
+
+                var buffer: [fs.MAX_NAME_BYTES]u8 = undefined;
+                const target_path = try entry.dir.readLink(entry.basename, buffer[0..]);
+                cwd.symLink(target_path, dest_path, .{}) catch |err| switch (err) {
+                    error.PathAlreadyExists => {},
```

Comment:
In the same way that installation of files ensures that the contents are the same even if the file already exists, this code needs to make sure that the already-installed symlink has the correct target. Or, if it is not a symlink but a file, it should be removed and replaced with a symlink. Furthermore, make sure that the vice versa case is handled when copying a file. If there is already a symlink there, it should be removed and replaced with the file. I am not sure whether `fs.Dir.updateFile` handles this case or not.